appinventor/components/src/com/google/appinventor/components/runtime/Component.java

```diff
@@ -128,23 +128,6 @@
   static final int TOAST_LENGTH_SHORT = 0;
   static final int TOAST_LENGTH_LONG = 1;
 
-  /*
-   * Screen direction constants.
-   * Observe that opposite directions have the same magnitude but opposite signs.
-   */
-  static final int DIRECTION_NORTH = 1;
-  static final int DIRECTION_NORTHEAST = 2;
-  static final int DIRECTION_EAST = 3;
-  static final int DIRECTION_SOUTHEAST = 4;
-  static final int DIRECTION_SOUTH = -1;
-  static final int DIRECTION_SOUTHWEST = -2;
-  static final int DIRECTION_WEST = -3;
-  static final int DIRECTION_NORTHWEST = -4;
-  // Special values
-  static final int DIRECTION_NONE = 0;
-  static final int DIRECTION_MIN = -4;
-  static final int DIRECTION_MAX = 4;
```

Comment:
I think these should be fine to remove because they're private (correct?).

Other constants like the the ones for map features have to stay b/c they're public.

Reply:
I think that these are *not* removable. Any other Java class definition can import this interface, and use any constant within it in a qualified way (such as Component.DIRECTION_NORTH), or, if the class is declared to implement the interface, can use an unqualified reference (just DIRECTION_NORTH). Since extension developers could do this (I think?), these are not removable.

Reply:
Sweet! I'll add them back in :D

Reply:
Done!